### ISSUE TITLE
False positive: "Null assignment can be safely removed" on an untyped static property that overrides a non-null parent default

### ISSUE THREAD

**Issue by abodnar** [kalessil/phpinspectionsea#2016]
## Summary

`SenselessNullAssignment` flags `= null` as removable on an untyped static property that redeclares a parent property whose default is non-null. Removing the `= null` here isn't safe. The child would go back to inheriting the parent's non-null default instead of `null`, so the behavior quietly changes.

This looks like a different case from the typed-property false positives in #1380 and #1772. Those were about typed properties needing explicit initialization. This one is untyped, and it's about inheritance shadowing, so I don't think the existing type-based handling covers it.

## Minimal reproduction

```php
<?php

class ParentResource
{
    // Non-null default in the parent.
    public static $wrap = 'data';
}

class ChildResource extends ParentResource
{
    // Overrides the inherited 'data' with null to disable wrapping.
    // EA flags this `= null` as "can be safely removed", but removing it
    // makes ChildResource inherit 'data' again, which changes behavior.
    public static $wrap = null;
}
```

## Expected

No warning on `ChildResource::$wrap`. The `= null` is needed and it shadows the parent's non-null default. Leaving it off doesn't give you `null`, it gives you `'data'` again.

## Actual

`[EA] Null assignment can be safely removed. Define null in annotations if it's important.` shows on `public static $wrap = null;`, and the quick-fix removes the initializer, which changes what the code does at runtime.

## Why it matters

This is the pattern Laravel's API resources use to turn off the `"data"` envelope. `Illuminate\Http\Resources\Json\JsonResource` declares `public static $wrap = 'data';`, and subclasses set `public static $wrap = null;` to disable wrapping (same as `JsonResource::withoutWrapping()`). The property is untyped in Laravel 11, 12, and 13, so this won't get fixed by the framework typing the property. Any resource that disables wrapping hits this.

## Likely cause

I think the inspection treats `= null` as redundant based on the local declaration, since an untyped uninitialized property is already `null`. It doesn't look at the parent's initializer. For a property that overrides a non-null parent default, that local view is wrong.

## Environment

- PHP Inspections (EA Extended) `6.3.0`
- PHP 8.3
- Reproducible with plain PHP classes, no framework needed (see above)
